### ISSUE TITLE
Investigate enabling features in regex crate to reduce compile times

### ISSUE THREAD

**Issue by alamb** [apache/arrow-rs#51]
*Note*: migrated from original JIRA: https://issues.apache.org/jira/browse/ARROW-6482

The regex crate recently added a feature flag to reduce compile times and binary size if certain unicode related features are not needed.  We should investigate using this feature.

**Comment by alamb** [comment]
Comment from Krisztian Szucs(kszucs) @ 2019-09-17T11:47:55.906+0000:
<pre>[~paddyhoran] may we postpone this after 0.15?</pre>

Comment from Paddy Horan(paddyhoran) @ 2019-09-18T02:25:23.049+0000:
<pre>Yep, will change.</pre>

Comment from Neville Dipale(nevi_me) @ 2020-06-01T11:31:39.865+0000:
<pre>[~paddyhoran] I've briefly looked at this, and it seems like the only feature we'd want to remove would be Unicode support ([https://docs.rs/regex/1.3.9/regex/#crate-features]). </pre>

Comment from Paddy Horan(paddyhoran) @ 2020-06-01T15:05:34.773+0000:
<pre>Yep, it's the unicode features I had in mind.</pre>

**Comment by sdf-jkl** [comment]
@Jefffrey @Rich-T-kid does anyone have issues with arrow-rs compile time?

_If_ needed we can add a compile time benchmark first. DataFusion already has one -- https://github.com/apache/datafusion/blob/main/benchmarks/README.md#compile-profile-benchmark

Turning off unicode support for regex didn't change the compile time on my machine, but this is not a real evidence.

Regex unicode support is used in `arrow-string` and I see no reason to reduce the regex capabilities or gate them behind a feature flag, unless it makes sense for the end user. 

> unless it makes sense for the end user. 

This seems like a reasonable idea though.

**Comment by Jefffrey** [comment]
yeah i dont have much context on this either, havent heard anything relating to this. if someone has time we could do a little investigation to see if its worth reducing the binary size + compile time per

- https://docs.rs/regex/latest/regex/#opt-out-of-unicode-support

at most we'd just add a new unicode feature thats on by default anyway 🤔

**Comment by alamb** [comment]
given how old this is, I think we should close it as unactionable. We can reopen if we have some idea of what we would do
